### ISSUE TITLE
Linux: dead keys still not working in some distributions/configurations

### ISSUE THREAD

**Issue by pdenapo** [zed-industries/zed#12089]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After #9207 was closed, dead-keys are not working in some Linux distributions/configurations.
For instance, type ` followed by a in a Spanish keyboard. You expect to get á

I have tested some. For instance, it does work in Linuxmint 21.3 (based on ubuntu 22.04) with
mate desktop. But they do not work in Debian 12 (bookwom) using Cinnamon, or in Endeavour using Xfce

I've got the following message at the start of Zed, and I guess it might be related to the cause of the problem 

[2024-05-21T11:05:51-03:00 ERROR gpui::platform::linux::x11::client] XIM Event Sender dropped

Please ask me any info that you think that it might help.
  

### Environment

For instance my endeavours OS running on a virtualbox virtual machine reports

uname -a 
Linux pablo-virtualbox 6.9.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 17 May 2024 16:56:38 +0000 x86_64 GNU/Linux

My keyboard configuration reported by

setxkb -print -verbose 10 

Setting verbose level to 10
locale is C
Trying to load rules file ./rules/evdev...
Trying to load rules file /usr/share/X11/xkb/rules/evdev...
Success.
Applied rules from evdev:
rules:      evdev
model:      pc105
layout:     latam
Trying to build keymap using the following components:
keycodes:   evdev+aliases(qwerty)
types:      complete
compat:     complete
symbols:    pc+latam+inet(evdev)
geometry:   pc(pc105)
xkb_keymap {
	xkb_keycodes  { include "evdev+aliases(qwerty)"	};
	xkb_types     { include "complete"	};
	xkb_compat    { include "complete"	};
	xkb_symbols   { include "pc+latam+inet(evdev)"	};
	xkb_geometry  { include "pc(pc105)"	};
};

**Comment by XDeme1** [comment]
Do you have any Input Method installed? I think in X11 you need to have something like fcitx5 installed to work in Zed right now, in Wayland it is supposed to work fine.

**Comment by jansol** [comment]
Dead keys should work without an input method via the xkb composition API that was also added in #11712.

That is assuming that your build of Zed is from a recent enough commit to contain those changes.

**Comment by jansol** [comment]
> it would be nice to have something like Zed --version reporting the git commit that was used

`zed: copy system specs into clipboard` reports that for git builds.

**Comment by pdenapo** [comment]
Great!

Zed: v1.0.0 (Zed Dev 410c46a5517e397890580c21dab13575e7839d5a) OS: Linux 1.0.0 Memory: 7.7 GiB Architecture: x86_64

It would be nice that a switch like zed --version prints this, because this would not depend on 
GPUI to start sucessfully! (and thus, it might be help full in other bug reports)
